metadata-ingestion/src/datahub/ingestion/source/sql/two_tier_sql_source.py

```diff
@@ -0,0 +1,95 @@
+import typing
+from typing import Any, Dict
+
+import pydantic
+from pydantic.fields import Field
+from sqlalchemy import create_engine, inspect
+from sqlalchemy.engine.reflection import Inspector
+
+from datahub.configuration.common import AllowDenyPattern
+from datahub.emitter.mcp_builder import PlatformKey
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.sql.sql_common import (
+    BasicSQLAlchemyConfig,
+    SQLAlchemySource,
+    logger,
+    make_sqlalchemy_uri,
+)
+
+
+class TwoTierSQLAlchemyConfig(BasicSQLAlchemyConfig):
+
+    database_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex patterns for databases to filter in ingestion.",
+    )
+    schema_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Deprecated in favour of database_pattern. Regex patterns for schemas to filter in ingestion. "
+        "Specify regex to only match the schema name. e.g. to match all tables in schema analytics, "
+        "use the regex 'analytics'",
+    )
+
+    @pydantic.root_validator()
+    def ensure_profiling_pattern_is_passed_to_profiling(
+        cls, values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        schema_pattern = values.get("schema_pattern")
+        if schema_pattern is not None:
+            values["database_pattern"] = schema_pattern
```

Comment:
I think this should only be overwritten if `database_pattern` is not set

Reply:
I was thinking this, but I count not found a way to detect if database_pattern is not set ( default value: allow all) vs the user actually setting it. Ideally now 'schema_pattern' should not be set by new recipes as it's deprecated. But let me know if we can do anything better here.

Reply:
I think it is fine to check if it is different than allow all then you don't override the schema pattern, and in both cases you send some warning.